### PR TITLE
Major edits to GraphQL docs section

### PR DESCRIPTION
**Pull request by agata-wit**
## Description
This PR improves all documentation pages sitting under: Tyk Stack > Tyk Gateway > GraphQL
- Review and changes to wording in all subpages
- Proper ordering of the subpages, so that they make more sense to the user
- Adding links to "dead end" pages
- Adding "Introspection queries" page with query examples
- Making sure all features are covered. In cases where there are differences between how you would do something in Dashboard vs OSS Gateway the differences are described
- Moving GQL subscriptions from Key concepts, to GQL because it just makes sense
- More examples and snippets to use OOTB

## Motivation and Context
This documentation has grown organically since we release GQL in 3.2 and never really had a proper audit and cleanup. Some information was outdated, some was missing, now it's fixed.

## Preview Link
https://deploy-preview-2459--tyk-docs.netlify.app/docs/nightly/graphql/

## Types of changes
- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [x] Something else (please add if needs merging to production or not)

## Checklist
- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I have added a preview link.

### CODE REVIEW

File: tyk-docs/content/graphql/introspection.md [hunk 1]
```diff
@@ -4,30 +4,33 @@ date: 2020-06-03
...
-an *introspection query* to the GraphQL server.
+A GraphQL server can provide information about its schema. This functionality is called **introspection** and is achievable by sending an **introspection query** to the GraphQL server.
 
-You may have seen *GraphiQL* or *GraphQL Playgrounds* providing a schema documentation and autocompletion. This is possible because those tools
-send a introspection query to the GraphQL server and use the response for providing those features.
-
-When **creating a GraphQL proxy** in the Tyk Dashboard an introspection query is used to fetch the schema from the GraphQL upstream and display it in the schema tab.
-
-In cases where upstream is protected and requires authorization to execute introspection query **Introspection headers** can be used to provide all necessary information. More information about Introspection headers can be found [on this page]({{< ref "/graphql/gql-headers">}})
+When [**creating a GraphQL proxy**]({{< ref "/graphql/creating-gql-api">}}) in Tyk Dashboard an introspection query is used to fetch the schema from the GraphQL upstream and display it in the schema tab.
```
Comment (pvormste) [inline tyk-docs/content/graphql/introspection.md]: this one is rendering as `<strong>creating a GraphQL proxy</strong>` in the preview

File: tyk-docs/content/graphql/introspection.md [hunk 2]
```diff
@@ -4,30 +4,33 @@ date: 2020-06-03
...
     parent: "GraphQL"
-weight: 1
+weight: 2
 aliases:
     - /graphql/introspection/
 ---
 
-A GraphQL server can provide information about its schema. This functionality is called *introspection* and is achievable by sending
-an *introspection query* to the GraphQL server.
+A GraphQL server can provide information about its schema. This functionality is called **introspection** and is achievable by sending an **introspection query** to the GraphQL server.
```
Comment (pvormste) [inline tyk-docs/content/graphql/introspection.md]: maybe we can also link to GraphQL spec here: https://spec.graphql.org/October2021/#sec-Introspection

File: tyk-docs/content/graphql/graphql-playground.md [hunk 3]
```diff
@@ -4,12 +4,20 @@ date: 2020-06-03
...
   main:
     parent: "GraphQL"
-weight: 2
+weight: 6
 aliases:
     - /graphql/graphql-playground/
 ---
 
-While creating or editing a GraphQL API, all the changes made, can be tested using the GraphiQL playground which sits under the `Playground` tab in `API Designer`.
+When you are creating or editing your GraphQL API, any change you make can be tested using Tyk Dashboard built-in GraphiQL Playground.
```
Comment (pvormste) [inline tyk-docs/content/graphql/graphql-playground.md:12]: Should we also mention that there is also an GraphQL API Playground that can be enabled via API definition? Also listing the differences maybe, like:
 - no debug logs
 - accessible via API path + `/playground` (default, but configurable)

Comment (agata-wit) [reply tyk-docs/content/graphql/graphql-playground.md:12]: Yes! I added this now!

File: tyk-docs/content/graphql/introspection.md (hunk 2 above)
Comment (agata-wit) [reply tyk-docs/content/graphql/introspection.md]: TBH I don't want to give too many external links, because they take the user away from our page and move them to someone else's and that's a risk they won't come back.
In some cases those external links are necessary but I think in this page we can do without it.

Comment (pvormste) [reply tyk-docs/content/graphql/introspection.md]: I do agree in general, but there are 2 reasons why I think it is a good idea to add it:
 1. Introspection is a concept that not many users are aware of, especially when using GraphQL for the first time. Our docs try to explain what GraphQL is so why not allow the user to read more about the introspection concept in the GraphQL Spec.
 2. The GraphQL spec is our baseline and people will try to figure out how compliant Tyk is with the spec. So it won't hurt to provide the references for faster access. In the end, the GraphQL spec is just the spec and users will need to come back to our docs because the spec doesn't explain how to work with Tyk - so I don't see a risk that users won't come back.

tl;dr: The spec is fundamental for us and having references to it can even increase awareness on our spec compliance :-)

Comment (agata-wit) [reply tyk-docs/content/graphql/introspection.md]: Good points. You convinced me :) Added!

**Review by pvormste — APPROVED** [review]

File: tyk-docs/content/graphql/creating-gql-api.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+```
+
+Congratulations - You have just created your first keyless GQL API, then protected it using Tyk!
+
+{{< tab_end >}}
+{{< tab_start "GQL API via Tyk Gateway API" >}}
+
+## Prerequisites
+
+In order to complete the next steps, you need to have the [Tyk Community Edition installed]({{< ref "tyk-oss-gateway" >}}).
```
Comment (letzya) [inline tyk-docs/content/graphql/creating-gql-api.md]:
```suggestion
In order to complete the next steps, you need to have the [Tyk OSS]({{< ref "tyk-oss-gateway" >}}) installed.
```

File: tyk-docs/content/graphql/complexity-limiting.md [hunk 5]
```diff
@@ -4,35 +4,16 @@ date: 2020-06-03
 menu:
   main:
     parent: "GraphQL"
-weight: 3
-aliases:
-    - /graphql/complexity-limiting/
```
Comment (letzya) [inline tyk-docs/content/graphql/complexity-limiting.md:9]: @agata-wit FYI - not need for an alias that is identical to the the file name and path

File: tyk-docs/content/getting-started/key-concepts/graphql-subscriptions.md [hunk 6]
```diff
@@ -5,15 +5,13 @@ tags: [""]
...
+weight: 11
 ---
 
-{{< toc >}}
+Tyk **natively** supports also GraphQL subscriptions, so you can expose your full range of GQL operations using Tyk Gateway. Subscriptions support was added in `v4.0.0` in which *graphql-ws* protocol support was introduced. 
 
-### Introduction
-
-Subscriptions is new functionality added in version `4.0.0` In simple terms subscriptions are a way to push data from the server to the clients that choose to listen to real-time messages from the server.
+Starting with Tyk `4.3.0` the number of supported subscription protocols has been extended.
```
Comment (letzya) [inline tyk-docs/content/getting-started/key-concepts/graphql-subscriptions.md]: Not sure I understand what does this mean? 
Also can you write` v4.3.0`

File: tyk-docs/content/getting-started/key-concepts/graphql-subscriptions.md [hunk 7]
```diff
@@ -25,381 +23,22 @@ In Tyk subscriptions are using the [WebSocket transport](https://developer.mozil
...
 Connections between client and Gateway currently only supports WebSockets/graphql-ws.
-See [GraphQL WebSockets]({{< ref "/graphql/graphql-websockets" >}}) for more information.
+See [GraphQL WebSockets]({{< ref "graphql/graphql-websockets" >}}) for more information.
 {{< /note >}}
+
 {{< note >}}
 **Note**  
 
-If the upstream subscription api is protected please enable the authentication via query params to pass header through.
+If the upstream subscription GQL API is protected please enable the authentication via query params to pass header through.
```
Comment (letzya) [inline tyk-docs/content/getting-started/key-concepts/graphql-subscriptions.md]:
```suggestion
If the upstream subscription GraphQL API is protected please enable the authentication via query params to pass the header through.
```

**Review by letzya — CHANGES_REQUESTED** [review]

File: tyk-docs/content/graphql/complexity-limiting.md (hunk 5 above)
Comment (agata-wit) [reply tyk-docs/content/graphql/complexity-limiting.md:9]: What's not needed? The weight? Or the alias itself?
The alias was already there, so I just left it as it was.

File: tyk-docs/content/getting-started/key-concepts/graphql-subscriptions.md (hunk 6 above)
Comment (agata-wit) [reply tyk-docs/content/getting-started/key-concepts/graphql-subscriptions.md]: I changed the sentence a bit to hopefully make it clearer and added the 'v' as well.

File: tyk-docs/content/graphql/complexity-limiting.md (hunk 5 above)
Comment (letzya) [reply tyk-docs/content/graphql/complexity-limiting.md:9]: yes, I just saw you added it in the past so was letting you know I removed it from this page.

**Review by letzya — APPROVED** [review]
